### ISSUE TITLE
Allow compile time purging to happen per application/module

### ISSUE THREAD

**Issue by josevalim** [elixir-lang/elixir#7095]
Requires #7046.

Mailing list discussion: https://groups.google.com/d/msgid/elixir-lang-core/606029ea-9855-44c6-ba94-69113ffed45c%40googlegroups.com?utm_medium=email&utm_source=footer

**Comment by whitfin** [comment]
@josevalim do you anticipate this working in the same way as levels currently work (meaning you can change an application's level at runtime assuming they're not purged out)? If so I wouldn't mind taking a shot at this once the other issue is ready & merged.

**Comment by josevalim** [comment]
The plan for now is only compile time.
-- 


*José Valimwww.plataformatec.com.br
<http://www.plataformatec.com.br/>Founder and Director of R&D*

**Comment by bitwalker** [comment]
What about purging by module as well? Both application and module are available at compile-time right? I think it provides a great deal of flexibility if you only want to silence a particularly noisy part of a dependency/application, rather than having to say "well module A is noisy, so I guess I just have to kill all logs from that application".

In the mailing list discussion, it was more about having control over the log level, so you could do things like choose to log more or less verbosely from a specific application or component, rather than just purging logs; is that not something you think is up for consideration, or are you thinking that's something for later?

**Comment by josevalim** [comment]
@bitwalker that's a great idea. We can likely generalize the filter for any compile time attribute.

**Comment by josevalim** [comment]
Closing in favor of #7777.
